helix-core/src/main/java/org/apache/helix/spectator/RoutingTableProvider.java

```diff
@@ -700,12 +1003,13 @@ public void queueEvent(NotificationContext context, ClusterEventType eventType,
       event.addAttribute(AttributeName.helixmanager.name(), context.getManager());
       event.addAttribute(AttributeName.changeContext.name(), context);
       queueEvent(event);
-
-      _monitor.increaseCallbackCounters(_eventQueue.size());
+      for (PropertyType propertyType : _monitorMap.keySet()) {
+        _monitorMap.get(propertyType).increaseCallbackCounters(_eventQueue.size());
```

Comment:
One question, if we have multiple state types for the customized state view, do we have multiple event count for this monitor? Wil that be bad?

Reply:
Yes this could potentially be not ideal and have some room to improvement. Based on this design (which we have one routerUpdater for all of the different types), we have one queue/RouterUpdater per RoutingTableProvider. Even here we are sharing the queue between them. Hence when we have hendle_event, we add queue size to all of them regarding the type and content of the queue. This is one of the drawbacks of having one RouterUpdater per RoutingTable. We can have TODO for improvement here maybe? What do you think?

Reply:
TODO is fine, but we need 2 items.
1. Split the monitor into 2. One is general router callback tracking. The other one is for each type of tracking. Then we have the right numbers.
2. We may need to add more complexity to the customized view monitor for each state type.

Reply:
Agree. Added the TODO with your points.